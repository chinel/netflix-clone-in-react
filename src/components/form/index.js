import React from "react";
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from "./styles/form";

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};
