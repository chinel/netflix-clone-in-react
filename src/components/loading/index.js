import React from "react";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner>
      <LockBody />
      <Picture src={`/images/users/${src}.png`}
    </Spinner>
  );
}
