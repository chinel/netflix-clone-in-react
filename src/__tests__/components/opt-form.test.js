import React from "react";
import { render } from "@testing-library/react";
import { OptForm } from "../../components";

describe("<OptForm />", () => {
  it("renders the <OptForm /> with populated data", () => {
    const { container, getByText, getByPlaceholderText } = render(
      <OptForm>
        <OptForm.Input placeholder="Email Address" />
        <OptForm.Button>Try it now</OptForm.Button>
        <OptForm.Break />
        <OptForm.Text>
          Ready to watch? Enter your email address to create or restart your
          membership.
        </OptForm.Text>
      </OptForm>
    );
  });
});
