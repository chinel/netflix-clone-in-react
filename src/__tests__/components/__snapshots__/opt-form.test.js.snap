// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptForm /> renders the <OptForm /> with populated data 1`] = `
<div
  class="opt-form__Container-brislo-0 doCxvu"
>
  <input
    class="opt-form__Input-brislo-1 btlmoz"
    placeholder="Email Address"
  />
  <button
    class="opt-form__Button-brislo-2 hZAoEs"
  >
    Try it now
    <img
      alt="Try Now"
      src="/images/icons/chevron-right.png"
    />
  </button>
  <div
    class="opt-form__Break-brislo-4 bJQdqk"
  />
  <p
    class="opt-form__Text-brislo-3 hMfrza"
  >
    Ready to watch? Enter your email address to create or restart your membership.
  </p>
</div>
`;
